File: app/src/main/java/com/reteno/sample/fragments/ecomevents/FragmentEcomEvents.java
package com.reteno.sample.fragments.ecomevents;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.navigation.fragment.NavHostFragment;

import com.reteno.sample.BaseFragment;
import com.reteno.sample.R;
import com.reteno.sample.databinding.FragmentEcomEventsBinding;

public class FragmentEcomEvents extends BaseFragment {

    private FragmentEcomEventsBinding binding;

    public FragmentEcomEvents() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentEcomEventsBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initListeners();
    }

    private void initListeners() {
        binding.btnProductViewed.setOnClickListener(v -> {
            navigateTo(R.id.ecomEvents_to_ProductViewed);
        });
        binding.btnProductCategoryViewed.setOnClickListener(v -> {
            navigateTo(R.id.ecomEvents_to_ProductCategoryViewed);
        });
        binding.btnProductAddedToWishlist.setOnClickListener(v -> {
            navigateTo(R.id.ecomEvents_to_ProductAddedToWishlist);
        });
        binding.btnCartUpdated.setOnClickListener(v -> {
            navigateTo(R.id.ecomEvents_to_CartUpdated);
        });
        binding.btnOrderCreated.setOnClickListener(v -> {
            navigateTo(R.id.ecomEvents_to_OrderCreated);
        });
        binding.btnOrderUpdated.setOnClickListener(v -> {
            navigateTo(R.id.ecomEvents_to_OrderUpdated);
        });
        binding.btnOrderDelivered.setOnClickListener(v -> {
            navigateTo(R.id.ecomEvents_to_OrderDelivered);

        });
        binding.btnOrderCancelled.setOnClickListener(v -> {
            navigateTo(R.id.ecomEvents_to_OrderCancelled);
        });
        binding.btnSearchRequest.setOnClickListener(v -> {
            navigateTo(R.id.ecomEvents_to_SearchRequest);
        });
    }

    private void navigateTo(int fragmentId) {
        NavHostFragment.findNavController(this).navigate(fragmentId);
    }
}

File: app/src/main/java/com/reteno/sample/fragments/ecomevents/FragmentEcomEvents.kt
package com.reteno.sample.fragments.ecomevents

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.fragment.NavHostFragment
import com.reteno.sample.BaseFragment
import com.reteno.sample.R
import com.reteno.sample.databinding.FragmentEcomEventsBinding

class FragmentEcomEvents : BaseFragment() {
    private var binding: FragmentEcomEventsBinding? = null
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentEcomEventsBinding.inflate(inflater, container, false)
        return binding!!.getRoot()
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initListeners()
    }

    private fun initListeners() {
        binding!!.btnProductViewed.setOnClickListener { navigateTo(R.id.ecomEvents_to_ProductViewed) }
        binding!!.btnProductCategoryViewed.setOnClickListener { navigateTo(R.id.ecomEvents_to_ProductCategoryViewed) }
        binding!!.btnProductAddedToWishlist.setOnClickListener { navigateTo(R.id.ecomEvents_to_ProductAddedToWishlist) }
        binding!!.btnCartUpdated.setOnClickListener { navigateTo(R.id.ecomEvents_to_CartUpdated) }
        binding!!.btnOrderCreated.setOnClickListener { navigateTo(R.id.ecomEvents_to_OrderCreated) }
        binding!!.btnOrderUpdated.setOnClickListener { navigateTo(R.id.ecomEvents_to_OrderUpdated) }
        binding!!.btnOrderDelivered.setOnClickListener { navigateTo(R.id.ecomEvents_to_OrderDelivered) }
        binding!!.btnOrderCancelled.setOnClickListener { navigateTo(R.id.ecomEvents_to_OrderCancelled) }
        binding!!.btnSearchRequest.setOnClickListener { navigateTo(R.id.ecomEvents_to_SearchRequest) }
    }

    private fun navigateTo(fragmentId: Int) {
        NavHostFragment.findNavController(this).navigate(fragmentId)
    }
}